apps/files/models.py

```diff
@@ -597,7 +597,7 @@ def save(self, *args, **kw):
         super(FileUpload, self).save()
 
     def add_file(self, chunks, filename, size):
-        filename = smart_str(filename)
+        filename = smart_str('{0}_{1}'.format(self.pk, filename))
```

Comment:
You need to make sure the `FileUpload` is already saved, or `self.pk` will be empty. Two solutions: either you add a `fu.save()` after https://github.com/mozilla/olympia/pull/769/files#diff-7962dc902d031e550d31c50d17f2ca20R618, or even better, add the following just above:
```
if not self.pk:
    self.save()
```